src/test/xint.cpp

```diff
@@ -493,7 +512,7 @@ TEST(xint, math1) {
 
 //const test_xint::detail::t_correct_int maxni = 0xFFFFFFFFFFFFFFFF; // max "normal integer" on this platform
 #define maxni 0xFFFFFFFFFFFFFFFFULL // max "normal integer" on this platform
-
+#define maxni_half 0x7FFFFFFFFFFFFFFFLL // max "normal integer" on this platform. signed, to allow - this value to reach the minimum signed value
```

Comment:
👍 

1) change both defines, to using 
const auto ......... = std numeric_limits
a) max - uint64_t 
b) min - int64_t

2) static assert check that first is equall to this 0xfff..ULL
b) and that min is equall to 8000000000000000 ULL